CONTRIBUTING.rst

```diff
@@ -380,6 +380,57 @@ pull requests from other contributors yourself after having reviewed
 them.
 
 
+Merge/squash guidelines
+-----------------------
+
+When a PR is approved and ready to be integrated to the ``main`` branch, one has the option to *merge* the commits unchanged, or *squash* all the commits into a single commit.
+
+Here are some guidelines on how to proceed, based on examples of a single PR commit history:
+
+1. Miscellaneous commits:
+
+   * ``Implement X``
+   * ``Fix test_a``
+   * ``Add myself to AUTHORS``
+   * ``!fixup Fix test_a``
```

Comment:
A more realistic example would be
```suggestion
   * ``fixup! Fix test_a``
```
(this is what `git commit --fixup` actually generates)